### ISSUE TITLE
`biom convert` changes integers to floats

### ISSUE THREAD

**Issue by gputzel** [biocore/biom-format#778]
When converting a BIOM 1.0 file to a TSV table, `biom convert` changes integers to floats.

For example, take the "minimal sparse OTU table" from [here](http://biom-format.org/documentation/format_versions/biom-1.0.html) and call it `minimal.json`.

Then

`biom convert -i minimal.biom -o otu_table.txt --to-tsv`

generates an OTU table `otu_table.txt` with floating-point values, despite the `matrix_element_type` being `int` in the original `biom` file.

**Comment by wasade** [comment]
To be fair, a TSV does not have datatypes and is pure ASCII, but I understand the issue being raised.

When we first defined the BIOM-Format, it made practical sense to encode the datatype as JSON is flat like TSV. When we shifted to the HDF5 based formats, we no longer had the need to encode the datatype as this was taken care of automatically on the definition of the datasets. For better or for worse now though, within the `Table` object, we implicitly cast to `float` to normalize the representation, and it's this implicit cast which is what's likely triggering the type change in the output. 

I'd welcome a pull request addressing this if there is interest. I wouldn't go as far as calling this a bug though, but more an unintended consequence of the format changes.
